Essentials/src/com/earth2me/essentials/utils/FormatUtil.java

```diff
@@ -71,17 +71,17 @@ public static String formatString(final IUser user, final String permBase, final
         if (user.isAuthorized(permBase + ".color")) {
             message = replaceColor(input, REPLACE_COLOR_PATTERN);
         } else {
-            message = stripColor(input, VANILLA_COLOR_PATTERN);
+            message = stripColor(stripColor(input, REPLACE_COLOR_PATTERN), VANILLA_COLOR_PATTERN);
```

Comment:
This doesn't look like anything to do with mail permissions.

Reply:
I see you addressed this in your overview. Are you sure this has no adverse effects?

Reply:
I'm not completely sure, I will have to do more testing. However, from the looks of it, it shouldn't do anything to the other stuff that uses `formatString(...)` or `fomatMessage(...)`.

Reply:
Classes affected by `formatString(...)` and `formatMessage(...)`:
- `mail`
- `msg` and `r`
- `me`
- `nick`
- SignChange listener

We can be sure that passing a String to `formatString(...)` is **certain** to colorize the message under the correct circumstances. Since I have not changed that, I can also assume that if I am stripping ampersand (`&`)-prefixed color codes, it is because the message would have been colorized anyways the user had the permission to do so; in other words, even if someone wrote something like:

```
/msg AgentTroll hi, I am messaging you&after I will put spaces in my messages
```

Their message would also be erroneously colored even if the codepath that I modified *was not* run.

Assuming also that the regex patterns and the usages of the two affected methods in `FormatUtil` are correct as well, I therefore have no reason to believe that those changes broke anything.